### PR TITLE
Added support for share service

### PR DESCRIPTION
**Pull request by katmsft**

### CODE REVIEW

File: sdk/storage/inc/shares/protocol/share_rest_client.hpp [hunk 1]
```diff
@@ -2557,18 +2558,41 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         {
           // Success
           ShareGetPropertiesResponse result;
-          result.Metadata = response.GetHeaders().at(Details::c_HeaderMetadata);
+
+          for (auto i = response.GetHeaders().lower_bound(Details::c_HeaderMetadata);
```
Comment (JinmingHu-MSFT) [inline sdk/storage/inc/shares/protocol/share_rest_client.hpp:2562]: `c_HeaderMetadata` should be defined as `x-ms-meta-`

File: sdk/storage/test/shares/share_client_test.cpp [hunk 2]
```diff
@@ -54,4 +66,120 @@ namespace Azure { namespace Storage { namespace Test {
...
+      EXPECT_EQ(metadata2, result);
+    }
+  }
+
+  TEST_F(FileShareClientTest, ShareQuota)
+  {
+    const int32_t quota32GB = 32;
+    const int32_t quota64GB = 64;
+    const int32_t quota5120GB = 5120;
+    const int32_t quotaInvalid = 5121;
```
Comment (JinmingHu-MSFT) [inline sdk/storage/test/shares/share_client_test.cpp:108]: if max quota is changed in the future, you'll have to fix this test case.

Comment (JinmingHu-MSFT) [reply sdk/storage/test/shares/share_client_test.cpp:108]: and you're testing server behavior

File: sdk/storage/inc/shares/protocol/share_rest_client.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/inc/shares/protocol/share_rest_client.hpp:2562]: Swagger uses `"x-ms-meta"` as the name of metadata. I think it's better to align.

File: sdk/storage/test/shares/share_client_test.cpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/test/shares/share_client_test.cpp:108]: Sure, I'll remove this.

**Pull request merged**